### PR TITLE
Add a patch to disable mpi4py imports with env var

### PR DESCRIPTION
**Pull request by xylar**
This is useful when working with `cdms2` on HPC login nodes, where `mpi4py` is in the environment because it is needed for compute nodes.

This PR also updates source location and license name.

Checklist
* [x] Used a fork of the feedstock to propose changes
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/conda_smithy.html#how-to-re-render ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by xylar** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -5,36 +5,40 @@ package:
...
       finity differencign.  Leverage the cores on the computer and is MPI ready.
-  dev_url: https://github.com/UV-CDAT/distarray
+  dev_url: https://github.com/CDAT/distarray
 
 extra:
   recipe-maintainers:
     - dnadeau4
     - doutriaux1
+    - xylar
+    - jasonb5
```
Comment (xylar) [inline recipe/meta.yaml:44]: @jasonb5, are you willing to be a maintainer?

File: recipe/0001-add-env-flag-to-disable-mpi4py.patch [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+@@ -2,11 +2,18 @@
+ Distributed array
+ """
+ __all__ = ["mvMultiArrayIter"]
+-from .mvMultiArrayIter import MultiArrayIter
++from .mvMultiArrayIter import MultiArrayIter
++
++import os
++
++mpi_disabled = os.environ.get("CDMS_NO_MPI", "False").lower() == 'true'
```
Comment (xylar) [inline recipe/0001-add-env-flag-to-disable-mpi4py.patch:13]: @jasonb5, I didn't crate a `util` module and `getenv_bool` function in this case, but could do so if you prefer.

**Comment by xylar** [comment]
Still running into trouble with `e3sm_diags` even with this patch.  Switching to draft until I figure more out.

**Comment by xylar** [comment]
With this patch, my tests are now working.  See https://github.com/conda-forge/cdms2-feedstock/pull/69#issuecomment-857761021

**Comment by xylar** [comment]
@jasonb5, could you unofficially review this when you have a chance?

**Comment by xylar** [comment]
@jasonb5, if you just take a glance at this, I'll ping the current maintainers to merge.

File: recipe/meta.yaml (hunk 1 above)
Comment (jasonb5) [reply recipe/meta.yaml:44]: @xylar I'm fine being a maintainer.

**Review by jasonb5 — APPROVED** [review]
LGTM

**Comment by xylar** [comment]
@dnadeau4 or @doutriaux1, would you please merge this?  Thanks very much!

**Comment by xylar** [comment]
@doutriaux1, thanks a lot!

**Pull request merged**